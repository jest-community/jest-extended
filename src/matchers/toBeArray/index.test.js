import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('.toBeArray', () => {
  it('passes when given an array', () => {
    expect([]).toBeArray();
  });

  it('fails when not given an array', () => {
    expect(() => expect(false).toBeArray()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeArray', () => {
  each([[false], [true], [0], [{}], [() => {}], [undefined], [null], [NaN]]).it(
    'passes when not given an array: %s',
    given => {
      expect(given).not.toBeArray();
    }
  );

  it('fails when given an array', () => {
    expect(() => expect([]).not.toBeArray()).toThrowErrorMatchingSnapshot();
  });
});
