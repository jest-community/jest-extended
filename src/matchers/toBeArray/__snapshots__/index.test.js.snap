// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeArray fails when given an array 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeArray(</><dim>)</>

Expected value to not be an array received:
  <red>[]</>"
`;

exports[`.toBeArray fails when not given an array 1`] = `
"<dim>expect(</><red>received</><dim>).toBeArray(</><dim>)</>

Expected value to be an array received:
  <red>false</>"
`;
